### PR TITLE
Fix sessions button

### PR DESCRIPTION
**Pull request by vansh284**
### Description

This PR aims to resolve https://github.com/source-academy/frontend/issues/2422. As mentioned by @shenyih0ng this issue is related to the switch to `AnchorButton`. Blueprintjs recommends the use of `AnchorButton` for tooltips over disabled buttons and so is required to resolve https://github.com/source-academy/frontend/issues/2417. However `AnchorButton` does not support submit buttons. 

The proposed solution is to use `AnchorButton` to render the `ControlButton` component only for disabled buttons. Buttons that are not disabled continue to use the regular Blueprintjs `Button`. Since `AnchorButton` and `Button` look the same, this shouldn't pose any problems. For reference https://blueprintjs.com/docs/#core/components/button.

Another possible and 'hacky' fix would be wrapping the `AnchorButton` with a HTML 'submit' button (refer to my first commit) but I prefer the current fix,

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### Checklist

- [x] I have tested this code

### CODE REVIEW

**Comment by vansh284** [comment]
> Is there any reason we want to keep using `Button` instead of `AnchorButton` when `isDisabled` is false?


`Button` allows passing in the HTML button type attribute so we are able to use our `ControlButton` component to submit form data. This does not work on `AnchorButton`. This is what caused the bug in the join sessions button in the first place.
image

When `isDisabled` is true, no functionality is expected from the button, so we can switch to `AnchorButton` to allow the tooltips to function properly.

File: src/commons/ControlButton.tsx [hunk 1]
```diff
@@ -37,8 +37,26 @@ const ControlButton: React.FC<ControlButtonProps> = ({
...
+        type={buttonOptions.type}
+        onClick={onClick}
+        icon={!buttonOptions.iconOnRight && iconElement}
+        rightIcon={buttonOptions.iconOnRight && iconElement}
+      >
+        {label}
+      </AnchorButton>
+    );
+  }
+
```
Comment (RichDom2185) [inline src/commons/ControlButton.tsx:57]:
```suggestion
  // Refer to #2417 and #2422 for why we conditionally
  // set the button component. See also:
  // https://blueprintjs.com/docs/#core/components/button
  const ButtonComponent = isDisabled ? AnchorButton : Button;
```

File: src/commons/ControlButton.tsx [hunk 2]
```diff
@@ -37,8 +37,26 @@ const ControlButton: React.FC<ControlButtonProps> = ({
...
   return (
-    <AnchorButton
+    <Button
```
Comment (RichDom2185) [inline src/commons/ControlButton.tsx:59]:
```suggestion
    <ButtonComponent
```

File: src/commons/ControlButton.tsx [hunk 3]
```diff
@@ -50,7 +68,7 @@ const ControlButton: React.FC<ControlButtonProps> = ({
       rightIcon={buttonOptions.iconOnRight && iconElement}
     >
       {label}
-    </AnchorButton>
+    </Button>
```
Comment (RichDom2185) [inline src/commons/ControlButton.tsx:71]:
```suggestion
    </ButtonComponent>
```

**Review by RichDom2185 — CHANGES_REQUESTED** [review]
Let's try to reduce duplicated code here.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**